Misc/NEWS.d/next/Library/2023-01-01-23-57-00.gh-issue-89727.ojedHN.rst

```diff
@@ -0,0 +1 @@
+Simplify and optimize os.walk by using isinstance checks to check the top of the stack.
```

Comment:
```suggestion
Simplify and optimize :func:`os.walk` by using :func:`isinstance` checks to check the top of the stack.
```
We could also skip the NEWS entry since this changes some code that was never released.